grails-app/services/org/pih/warehouse/data/ProductSupplierDataService.groovy

```diff
@@ -141,9 +141,27 @@ class ProductSupplierDataService {
     }
 
     def getOrCreateNew(Map params) {
-        def productSupplier = params.productSupplier ? ProductSupplier.get(params.productSupplier) : null
+        def productSupplier
+        if (params.productSupplier) {
+            productSupplier = params.productSupplier ? ProductSupplier.get(params.productSupplier) : null
+        } else {
+            productSupplier = ProductSupplier.createCriteria().list {
+                eq("product", params.product)
+                if (params.supplier && params.supplierCode && params.product) {
+                    eq("supplier", params.supplier)
+                    eq("supplierCode", params.supplierCode.replaceAll('[,-]',''))
+                    // TODO we need to find a way to replace dashes in attribues, solutions below are not working
```

Comment:
I think this could be done in beforeInsert/beforeUpdate. We could strip these characters out before saving to the database. That might not be what Kelsey wants so I'll see if there's another way of doing this in SQL/HQL. You also need to make the filter case-insensitive so a toLowerCase/toUpperCase on each side.